idc_index/index.py

```diff
@@ -502,6 +465,364 @@ def get_viewer_URL(
+        Returns:
+            total_size (float): The total size of all series in the manifest file.
+            endpoint_to_use (str): The endpoint URL to use (either AWS or GCP).
+            temp_manifest_file(Path): Path to the temporary manifest file for downstream steps
+        Raises:
+            ValueError: If the manifest file does not exist, if any URL in the manifest file is invalid, or if any URL is inaccessible in both AWS and GCP.
+            Exception: If the manifest contains URLs from both AWS and GCP.
+        """
+
+        # Read the csv file
+        manifest_df = pd.read_csv(
+            manifestFile, comment="#", skip_blank_lines=True, header=None
+        )
+
+        # Rename the column
+        manifest_df.columns = ["manifest_cp_cmd"]
+
+        # Apply the function to the 'manifest_cp_cmd' column
+        manifest_df["manifest_crdc_instance_uuid"] = manifest_df[
+            "manifest_cp_cmd"
+        ].apply(self._extract_crdc_instance_uuid_from_manifest_s3_url)
+
+        # Remove 'cp ' at the beginning and ' .' at the end of the 'manifest_cp_cmd'
+        manifest_df["s3_url"] = (
+            manifest_df["manifest_cp_cmd"]
+            .str.replace(r"^cp ", "", regex=True)
+            .str.replace(r" \.$", "", regex=True)
+        )
+        index_df_copy = self.index
+        index_df_copy["index_crdc_instance_uuid"] = index_df_copy[
+            "series_aws_url"
+        ].apply(self._extract_crdc_instance_uuid_from_series_aws_url)
+        # Perform left join on client.index using crdc_instance_uuid
+        merged_df = manifest_df.merge(
+            index_df_copy,
+            left_on="manifest_crdc_instance_uuid",
+            right_on="index_crdc_instance_uuid",
+            how="left",
+        )
+
+        # Create 'crdc_instance_uuid_found' column
+        merged_df["crdc_instance_uuid_found"] = merged_df[
+            "index_crdc_instance_uuid"
+        ].notna()
+
+        # Check if crdc_instance_uuid is found in the index
+        if not all(merged_df["crdc_instance_uuid_found"]):
+            missing_manifest_cp_cmds = merged_df.loc[
+                ~merged_df["crdc_instance_uuid_found"], "manifest_cp_cmd"
+            ]
+            missing_manifest_cp_cmds_str = f"The following manifest copy commands do not have any associated series in the index: {missing_manifest_cp_cmds.tolist()}"
+            raise ValueError(missing_manifest_cp_cmds_str)
+
+        # Create 'endpoint' column
+        merged_df["endpoint"] = merged_df.apply(
+            lambda row: aws_endpoint_url
+            if row["crdc_instance_uuid_found"]
+            and row["s3_url"] == row["series_aws_url"]
+            else gcp_endpoint_url,
+            axis=1,
+        )
+
+        # Check if there are more than one endpoints
+        if len(merged_df["endpoint"].unique()) > 1:
+            raise ValueError(
+                "The manifest contains URLs from both AWS and GCP. Please use only one provider."
```

Comment:
I may as well have a bogus bucket ID - this message will not be accurate in the general case and may be misleading.

Reply:
checking the first url with `s5cmd ls` if endpoint is not aws. All unusual errors are captured by a generic error.

Reply:
Reverted to printing a more specific error that catches both possibilities "Either GCS bucket path is invalid or manifest has a mix of GCS and AWS urls. If so, please use urls from one provider only"
"